### ISSUE TITLE
Breaking API changes for tslint integrators are not documented in the v5 release notes

### ISSUE THREAD

**Issue by egamma** [palantir/tslint#2439]
There are several breaking changes in the tslint API for integrators that are not documented in the release notes. While typescript helps to find the breaking changes, it would be helpful when the breaking changes are documented in the release notes.

These are the breaking changes I've discoverd when adding support for tslint 5

- `LintResult.failureCount` -> `LintResult.errorCount`
-  `class Fix` -> `type Fix = Replacement | Replacement[]`

**Comment by adidahiya** [comment]
Hi @egamma, sorry about that -- I've gone ahead and added those breaks to the [5.0.0 release notes](https://github.com/palantir/tslint/releases/tag/5.0.0). Let us know if there are others. Also, we're here to help and would be happy to review migration PRs for v5 (we use vscode-tslint a lot too!)

**Comment by egamma** [comment]
@adidahiya no problem and glad to hear that you use vscode-tslint, 

I've just published a new version of `vscode-tslint` that supports tslint 5.0. This is the commit with the changes https://github.com/Microsoft/vscode-tslint/commit/bd698afd5d217f43ae29d2a3c0ff8f9137778de8.

**Comment by donaldpipowitch** [comment]
> Hi @egamma, sorry about that -- I've gone ahead and added those breaks to the 5.0.0 release notes.

Just stumbled upon `LintResult.failureCount -> LintResult.errorCount`. I just looked at `CHANGELOG.md` before. Maybe it would be good to have just one source of truth for changelogs? (If you decide to use the release notes I'd keep the `CHANGELOG.md` thou just to point to the release notes.)

One disadvantage of release notes: they aren't searchable. I found this issue, because I searched for `failureCount`. `CHANGELOG.md` would be searched, too.
